packages/trader/src/Modules/Trading/Components/Form/TradeParams/Turbos/barrier-selector.jsx

```diff
@@ -3,14 +3,53 @@ import React from 'react';
 import { Localize, localize } from '@deriv/translations';
 import { connect } from 'Stores/connect';
 import { CSSTransition } from 'react-transition-group';
-import { Icon, Text, ThemedScrollbars } from '@deriv/components';
+import { Button, Icon, MobileDialog, Text, ThemedScrollbars, Popover } from '@deriv/components';
 import { isMobile } from '@deriv/shared';
 import Fieldset from 'App/Components/Form/fieldset.jsx';
 import BarriersList from './barriers-list.jsx';
 
 const BarrierSelector = ({ barrier_1, onChange, setHoveredBarrier, turbos_barrier_choices }) => {
     const [selected_barrier, setSelectedBarrier] = React.useState(barrier_1);
     const [is_barriers_table_expanded, setIsBarriersTableExpanded] = React.useState(false);
+    const [is_mobile_tooltip_visible, setIsMobileTooltipVisible] = React.useState(false);
+    const [is_button_blocked, setIsButtonBlocked] = React.useState(true);
+
+    const toggleMobileTooltip = () => setIsMobileTooltipVisible(!is_mobile_tooltip_visible);
+
+    const toggleBarriersTable = () => {
+        setIsButtonBlocked(true);
+        setIsMobileTooltipVisible(false);
+        setIsBarriersTableExpanded(!is_barriers_table_expanded);
+        setSelectedBarrier(barrier_1);
+    };
+
+    const onBarrierClick = barrier => {
+        setHoveredBarrier(null);
+        setSelectedBarrier(barrier);
+        setIsButtonBlocked(barrier === barrier_1);
+        if (!isMobile()) {
```

Comment:
On mobile version we don't send new value of `barrier_1` immediately after user taped on it; instead, we send that data only after the submit button was clicked (`onButtonClick` function). [Design link](https://www.figma.com/file/xqAQd87EYFFwIqPVKYbeYh/Turbos?node-id=6054%3A609338&t=En0rsOvHdBREebqL-0)